### PR TITLE
plugin API method to close_in_tray

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds a new method to all plugin user APIs to close the item in the tray (useful for demos and notebook workflows, especially when the plugin in question has live-previews enabled).  By default, it does not close the tray itself, but accepts an optional argument to close the entire tray.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
Actually for both open and close, should they be no-op if the requested item is already opened/closed? Then we don't have to loop over stuff for no reason, or is that too hard to find out before hand or it doesn't matter because the underlying call will already be a no-op?

**Comment by kecnry** [comment]
I _suspect_ it would cost just as much to add the check before updating the traitlet as it does to set the traitlet to the same value that it currently is, but if anyone wants to test that theory and add the test if its cheaper, feel free!

File: CHANGES.rst [hunk 1]
```diff
@@ -17,6 +17,8 @@ New Features
 
 - Reorder viewer and layer settings in Plot Options. [#2543, #2557]
 
+- Plugin APIs now include a ``close_in_tray()`` method. [#2562]
```
Comment (pllim) [inline CHANGES.rst:20]:
```suggestion
- Plugin APIs now include a ``close_in_tray()`` method. [#2562]
```

**Review by pllim — APPROVED** [review]
Seems to work. Very convenient to have indeed. Thanks!

**Pull request merged**